pkg/orchestrator/gitHubActions.go

```diff
@@ -1,18 +1,65 @@
 package orchestrator
 
 import (
+	"bytes"
+	"context"
+	"fmt"
+	"io"
+	"net/http"
 	"strings"
+	"sync"
 	"time"
 
+	piperHttp "github.com/SAP/jenkins-library/pkg/http"
 	"github.com/SAP/jenkins-library/pkg/log"
+
+	"golang.org/x/sync/errgroup"
+	"golang.org/x/sync/semaphore"
 )
 
-type GitHubActionsConfigProvider struct{}
+var httpHeader = http.Header{
+	"Accept": {"application/vnd.github+json"},
+}
+
+type GitHubActionsConfigProvider struct {
+	client piperHttp.Client
+}
+
+type Job struct {
```

Comment:
Could we merge all these structs in `GitHubActionsConfigProvider`?
Would this make sense?

Reply:
I don’t think it makes sense. These structs are like a helper, there help to get stages ID, and rely on IDs we can get full logs. IDs are used by only one method currently. 
Probably in the future, structs will be merged to `GitHubActionsConfigProvider`, but for now, let’s leave it as it is.

Reply:
I'd change it now (if applicable). There wont be a later...

Reply:
I currently don't see reusability purposes of the information that these structs will contain, so adding them to `GitHubActionsConfigProvider` seems unnecessary.

Reply:
Ok, but maybe make them private as they are only used inside the orchestrator package, or?

Reply:
True

Reply:
Done [695934d](https://github.com/SAP/jenkins-library/pull/4250/commits/695934d155f70e6356873069ef1f1a8307e30e2d).